azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java

```diff
@@ -151,6 +157,39 @@ public static boolean isExecutableJar(File file) {
         }
     }
 
+    private static <T> Callable<T> getWrappedCallable(String prompt, Future<T> future) {
+        return () -> {
+            System.out.print(prompt);
+            try {
+                for (int i = 0; !future.isDone(); i++) {
+                    System.out.print(PENDING_STRING_LIST[i % 4]);
+                    Thread.sleep(500);
+                    System.out.print("\b\b\b");
+                }
+                return future.get();
+            } catch (InterruptedException e) {
+                future.cancel(true);
+                return null;
+            } finally {
+                System.out.println();
+            }
+        };
+    }
+
+    public static <T> T executeCallableWithPrompt(Callable<T> callable, String prompt, int timeOutInSeconds) {
+        final ExecutorService executorService = Executors.newFixedThreadPool(2);
```

Comment:
why 2 ?

Reply:
One for execute customer callable and another one is used for showing pending message